### ISSUE TITLE
Make limit the max number to return

### ISSUE THREAD

**Issue by philnorc** [Greenstand/treetracker-wallet-api#276]
At
GET /wallet/wallets?limit=n&start=n
GET /wallet/tokens?limit=n&start=n
GET /wallet/tokens/<token_id>/transactions?limit=n&start=n
GET /wallet/transfers?limit=n&start=n
GET /wallet/transfers/<transfer_id>/tokens?limit=n&start=n
GET /wallet/trust_relationships?limit=n&start=n

?limit=n&start=n produces unexpected results.
Issue 261 replaces "start, defaults to 1" with "offset, defaults to 0"

It seems we should also fix limit to mean the maximum number of items to return,
not the index of the last item to select from the source array.

In other words, replace JavaScript array.slice(start,limit) with
array.slice(start,(start+limit))

Why? Because as it works now, limit can be a negative number, which
is hard to understand if you don't know the length of the array.
And because if start>limit, it returns nothing:
?limit=4&start=5 returns []

And limit < 1 should return the same error as start < 1:
422: "limit" must be greater than or equal to 1.

**Comment by Kpoke** [comment]
#261
